packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -483,15 +483,9 @@ async function handler(
   if (eventType.seatsPerTimeSlot) {
     const booking = await prisma.booking.findFirst({
       where: {
-        OR: [
```

Comment:
Due to this OR condition, in case of reschedule , the check on timeslot was skipped.
Hence the reschedule scenario was considered as second seat booking, isFirstSeat was set to false.
So user availability check was skipped causing to book in nonavailability slots.

Timeslot is the key here to determine if same slot is being booked for second/multiple seats.
Tested for #15105 scenario as well